backend/eventloop/shared_resource_loop/sharedresourceloop_managedenv.go

```diff
@@ -787,6 +794,13 @@ func createNewClusterCredentials(ctx context.Context, managedEnvironment managed
 	log.Info("Created ClusterCredentials for ManagedEnvironment", clusterCredentials.GetAsLogKeyValues()...)
 
 	updateManagedEnvironmentConnectionStatus(&managedEnvironment, ctx, workspaceClient, metav1.ConditionTrue, ConditionReasonSucceeded, "", log)
+
+	// If an existing service account is used instead, we should verify the cluster credentials based on the provided token
```

Comment:
in the `!createNewServiceAccount` case, we likely want to call `verifyClusterCredentials` *before* creating the cluster credentials, e.g. moving it a few lines up above the call to `dbQueries.CreateClusterCredentials`

Reply:
I moved it up, and if the validation fails, it just logs the error.